react/training/src/redux/login/reducers.js

```diff
@@ -0,0 +1,17 @@
+import { actionTypes } from './actions';
+
+const initialState = {
+  userAuthenticated: false
+};
+
+export default function logInReducer(state = initialState, action) {
+  switch (action.type) {
+    case actionTypes.SUBMIT:
+      return { ...state, userAuthenticated: action.payload };
+    case actionTypes.LOG_IN_FAIL:
+      window.alert(action.payload);
```

Comment:
try to avoid using `window.alert...`. Just write your error in the screen with React components